src/formatters/source/pushResultFormatter.ts

```diff
@@ -132,33 +133,32 @@ export class PushResultFormatter extends ResultFormatter {
     }
     // "content" property of the bundles as a string
     const contentFilePathFromDeployedBundles = this.componentsFromFilenames(
-      // the .filter(isString) should ensure only strings are present, but TS isn't finding that
-      bundlesDeployed.map((fileResponse) => fileResponse.filePath as string)
+      bundlesDeployed.map((fileResponse) => fileResponse.filePath).filter(isString)
```

Comment:
SDR has a handler for this now in https://github.com/salesforcecli/plugin-source/pull/1081 but this eliminates the evil asssertion.  `filePath` is typed as `string | undefined` and we'll respect that.